### ISSUE TITLE
[stable/prometheus-operator] adding additionalPrometheusRulesMap results in Helm error

### ISSUE THREAD

**Issue by LukeBalizetPatriot** [helm/charts#23057]
**Describe the bug**
When added to `values.yaml`, `additionalPrometheusRulesMap` does not allow `stable/prometheues` to upgrade/install

**Version of Helm and Kubernetes**:
```
Client Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.10-eks-bac369", GitCommit:"bac3690554985327ae4d13e42169e8b1c2f37226", GitTreeState:"clean", BuildDate:"2020-02-21T23:37:18Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.8-eks-e16311", GitCommit:"e163110a04dcb2f39c3325af96d019b4925419eb", GitTreeState:"clean", BuildDate:"2020-03-27T22:37:12Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
```

**Which chart**:
`stable/prometheus-operator`

**What happened**:
Adding to the `additionalPrometheusRulesMap` map results in the following error:
```
Error: UPGRADE FAILED: template: prometheus-operator/templates/_helpers.tpl:88:16: executing "prometheus-operator.namespace" at <.Values.namespaceOverride>: nil pointer evaluating interface {}.namespaceOverride
```

**What you expected to happen**:
For the Helm release to update/install successfully and add the additional Prometheus rules

**How to reproduce it** (as minimally and precisely as possible):
This is my current configuration in `values.yaml`
```
# values.yaml

additionalPrometheusRulesMap:
  blackbox-exporter:
    groups:
      - name: blackbox-exporter
        rules:
          - alert: Non2xxStatus
            expr: (probe_success != 0) + on(instance) (probe_http_status_code)
            for: 1m
            labels:
              severity: critical
            annotations:
              message: "{{ $labels.instance }} is returning a {{ $value }} status code"
```
No matter what I add to `additionalPrometheusRulesMap`,  I still receive the same error

**Anything else we need to know**:
The release works without `additionalPrometheusRulesMap`, but will not work with it

**Comment by davejhahn** [comment]
I'm having what I think is a similar problem.  I've created a ConfigMap with additional rules and when I upgrade, passing the config map name, e.g. additionalPrometheusRulesMap=monitoring-rules, I get:

Error: UPGRADE FAILED: template: prometheus-operator/templates/prometheus/additionalPrometheusRules.yaml:6:57: executing "prometheus-operator/templates/prometheus/additionalPrometheusRules.yaml" at <.Values.additionalPrometheusRulesMap>: range can't iterate over monitoring-rules

`apiVersion: v1
kind: ConfigMap
metadata:
  name: monitoring-rules
  namespace: monitoring
data:
  kafka-monitoring.yaml: |
    groups:
    - name: kafka.rules
      rules:
      - alert: KafkaTopicsReplicas
        annotations:
          summary: "Kafka topics replicas (instance {{ $labels.instance }})"
          description: "Kafka topic in-sync partition\n  VALUE = {{ $value }}\n  LABELS: {{ $labels }}"
        expr: sum(kafka_topic_partition_in_sync_replica) by (topic) < 3
        for: 5m
        labels:
          severity: critical
      - alert: KafkaConsumersGroup
        annotations:
          summary: "Kafka consumers group (instance {{ $labels.instance }})"
          description: "Kafka consumers group\n  VALUE = {{ $value }}\n  LABELS: {{ $labels }}"
        expr: sum(kafka_consumergroup_lag) by (consumergroup) > 50
        for: 5m
        labels:
          severity: critical`

I'm pretty sure the YAML is formatted correctly, even tried pulling a rule from prometheus-prometheus-operator-prometheus-rulefiles-0 configmap and changing the names, same issue.

**Comment by davejhahn** [comment]
I was able to get this to work by passing in both --set for the values I was already passing and also -f with a file with the rules.  I did not realize you could pass a file and --set values, go figure   But, I would still like to understand how the additionalPrometheusRulesMap value is supposed to be used.   

I scrapped the ConfigMap after it became kind of clear that was not what it was looking for and used --set-file additionalPrometheusRulesMap=<filename> but could not ever get past the unable to iterate no matter what I tried.

**Comment by mio-g** [comment]
Had the same issue. I think it has to do with the new `namespaceOverride` feature. It was fix in version 8.15.11 . Try run `helm repo update`

**Comment by LukeBalizetPatriot** [comment]
@mio-g that did it. Thank you!
